lib/oban/notifier.ex

```diff
@@ -66,16 +56,17 @@ defmodule Oban.Notifier do
       end
   """
 
-  use GenServer
-
-  import Oban.Breaker, only: [open_circuit: 1, trip_circuit: 3]
-
-  alias Oban.{Config, Registry, Repo}
-  alias Postgrex.Notifications
+  alias Oban.{Config, Registry}
 
+  @type server :: GenServer.server()
   @type option :: {:name, module()} | {:conf, Config.t()}
   @type channel :: :gossip | :insert | :signal
 
+  @callback start_link([option]) :: GenServer.on_start()
+  @callback listen(server(), channels :: list(channel())) :: :ok
```

Comment:
These callbacks populate the docs for the notifier implementations. They need to describe what the function is doing a little, at least as a stub.